### PR TITLE
chore: remove web3x

### PR DESCRIPTION
**Pull request by menduz**
- removes (many) dependencies, including web3x
- uses ethereum-cryptography for ECDSA functions
- replaces tslint -> eslint
- replaces mocha+sinon+nyc -> jest
- migrates circleci -> github actions
- adds extra tests

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,34 +42,24 @@
   },
   "homepage": "https://github.com/decentraland/decentraland-crypto#readme",
   "dependencies": {
-    "eth-crypto": "^1.5.1",
-    "web3x": "^4.0.6"
```
Comment (nachomazzara) [inline package.json:45]: see you in hell 👿

Comment (menduz) [reply package.json:45]: can I assume that is an approved PR?

File: src/crypto.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  )
+}
+
+export function sign(privateKey: Uint8Array, hash: Uint8Array): string {
+  const sigObj = ecdsaSign(hash, privateKey)
+  const recoveryId = sigObj.recid === 1 ? '1c' : '1b'
+  return toHex(sigObj.signature) + recoveryId
+}
+
+// TODO: unit test
```
Comment (nachomazzara) [inline src/crypto.ts:73]: Oops

File: test/authenticator.spec.ts [hunk 3]
```diff
@@ -49,16 +43,14 @@ describe('Decentraland Crypto', function () {
       const result = await Authenticator.validateSignature(
         'message',
         chain,
-        new HttpProvider(
-          'https://mainnet.infura.io/v3/640777fe168f4b0091c93726b4f0463a'
-        )
+        new HTTPProvider('https://rpc.decentraland.org/mainnet')
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:46]: Can we use `https://rpc.decentraland.org/mainnet?project=dcl-crypto-test` so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 4]
```diff
@@ -143,17 +131,15 @@ describe('Decentraland Crypto', function () {
         authority,
         authLink,
         {
-          provider: new HttpProvider(
-            'https://mainnet.infura.io/v3/640777fe168f4b0091c93726b4f0463a'
-          ),
+          provider: new HTTPProvider('https://rpc.decentraland.org/mainnet'),
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:134]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 5]
```diff
@@ -166,12 +152,10 @@ describe('Decentraland Crypto', function () {
       const result = await Authenticator.validateSignature(
         'QmWyFNeHbxXaPtUnzKvDZPpKSa4d5anZEZEFJ8TC1WgcfU',
         chain,
-        new HttpProvider(
-          'https://mainnet.infura.io/v3/2c902c2e3b8947d3b34bba7ca48635fc'
-        )
+        new HTTPProvider('https://rpc.decentraland.org/mainnet')
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:155]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 6]
```diff
@@ -184,18 +168,16 @@ describe('Decentraland Crypto', function () {
       const result = await Authenticator.validateSignature(
         'QmNUd7Cyoo9CREGsACkvBrQSb3KjhWX379FVsdjTCGsTAz',
         chain,
-        new HttpProvider(
-          'https://mainnet.infura.io/v3/2c902c2e3b8947d3b34bba7ca48635fc'
-        ),
+        new HTTPProvider('https://rpc.decentraland.org/mainnet'),
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:171]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 7]
```diff
@@ -286,15 +264,13 @@ describe('Decentraland Crypto', function () {
       const result = await Authenticator.validateSignature(
         'QmbGrShBQs4XiuoTNX6znAvXNdqtub8DtXyaxdSTZbHLCu',
         chain,
-        new HttpProvider(
-          'https://mainnet.infura.io/v3/640777fe168f4b0091c93726b4f0463a'
-        )
+        new HTTPProvider('https://rpc.decentraland.org/mainnet')
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:267]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 8]
```diff
@@ -248,20 +228,18 @@ describe('Decentraland Crypto', function () {
       const result = await Authenticator.validateSignature(
         'QmUe3LmUJ4NACAKJzwQhn5rZVpLLSyBLWBmTSzJYEesDNx',
         chain,
-        new HttpProvider(
-          'https://mainnet.infura.io/v3/640777fe168f4b0091c93726b4f0463a'
-        )
+        new HTTPProvider('https://rpc.decentraland.org/mainnet')
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:231]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 9]
```diff
@@ -208,22 +190,20 @@ describe('Decentraland Crypto', function () {
         authority,
         authLink,
         {
-          provider: new HttpProvider(
-            'https://mainnet.infura.io/v3/640777fe168f4b0091c93726b4f0463a'
-          ),
+          provider: new HTTPProvider('https://rpc.decentraland.org/mainnet'),
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:193]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 10]
```diff
@@ -307,15 +283,11 @@ describe('Decentraland Crypto', function () {
       }
       try {
         await ECDSA_PERSONAL_EPHEMERAL_VALIDATOR(authority, authLink, {
-          provider: new HttpProvider(
-            'https://mainnet.infura.io/v3/640777fe168f4b0091c93726b4f0463a'
-          ),
+          provider: new HTTPProvider('https://rpc.decentraland.org/mainnet'),
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:286]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 11]
```diff
@@ -333,35 +305,27 @@ describe('Decentraland Crypto', function () {
       let result = await Authenticator.validateSignature(
         'message',
         chain,
-        new HttpProvider(
-          'https://mainnet.infura.io/v3/640777fe168f4b0091c93726b4f0463a'
-        )
+        new HTTPProvider('https://rpc.decentraland.org/mainnet')
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:308]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 12]
```diff
@@ -333,35 +305,27 @@ describe('Decentraland Crypto', function () {
...
       )
 
       // Since we are checking the ephemeral against 10 minutes ago, validation should pass
       result = await Authenticator.validateSignature(
         'message',
         chain,
-        new HttpProvider(
-          'https://mainnet.infura.io/v3/640777fe168f4b0091c93726b4f0463a'
-        ),
+        new HTTPProvider('https://rpc.decentraland.org/mainnet'),
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:319]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 13]
```diff
@@ -333,35 +305,27 @@ describe('Decentraland Crypto', function () {
...
+      expect(result.ok).toEqual(true)
     })
 
     it('should validate authChain', async function () {
-      const clock = sinon.useFakeTimers(0)
-      const provider = new HttpProvider(
-        'https://mainnet.infura.io/v3/640777fe168f4b0091c93726b4f0463a'
-      )
+      jest.useFakeTimers().setSystemTime(0)
+      const provider = new HTTPProvider('https://rpc.decentraland.org/mainnet')
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:328]: Can we use https://rpc.decentraland.org/mainnet?project=dcl-crypto-test so we can discriminate the usage?

File: test/authenticator.spec.ts [hunk 14]
```diff
@@ -495,7 +453,7 @@ describe('Decentraland Crypto', function () {
             '0xd73b0315dd39080d9b6d1a613a56732a75d68d2cef2a38f3b7be12bdab3c59830c92c6bdf394dcb47ba1aa736e0338cf9112c9eee59dbe4109b8af6a993b12d71b'
         }
       ]
-      expect(Authenticator.isValidAuthChain(chain)).to.be.equal(true)
+      expect(Authenticator.isValidAuthChain(chain)).toEqual(true)
     })
   })
 })
```
Comment (nachomazzara) [inline test/authenticator.spec.ts:459]: I know that it was that way before but maybe we can move the RPC URL to a constant at the top of the file.

**Review by nachomazzara — CHANGES_REQUESTED** [review]
Thanks @menduz 🎸

**Pull request merged**